core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2568,22 +2559,32 @@ private Type coerceToSingleType(StackableAstVisitorContext<Context> context, Str
         {
             // determine super type
             Type superType = UNKNOWN;
+
+            ListMultimap<Type, Expression> typeExpressions = ArrayListMultimap.create();
+            Set<Type> orderedTypes = new LinkedHashSet<>();
```

Comment:
use https://guava.dev/releases/23.0/api/docs/com/google/common/collect/LinkedHashMultimap.html instead

Reply:
It was trickier. LinkedHashMultimap doesn't allow duplicated values so I needed to wrap into NodeRef right away.